src/documentation/literal.rs

```diff
@@ -0,0 +1,443 @@
+            }
+        } else {
+            return Err(anyhow!("Missing prefix of literal: {}", rendered.as_str()));
+        };
+        let post = if let Some(captures) = SUFFIX_ERASER.captures(rendered.as_str()) {
+            // capture indices are 1 based, 0 is the full string
+            if let Some(suffix) = captures.get(captures.len() - 1) {
+                suffix.as_str().len()
+            } else {
+                return Err(anyhow!("Unknown suffix of literal"));
+            }
+        } else {
+            return Err(anyhow!("Missing suffix of literal: {}", rendered.as_str()));
+        };
+        let _scrap = |c: &'_ char| -> bool { c.is_whitespace() };
+
+        let (len, pre, post) = match rendered.len() {
+            len if len >= pre + post => (len - pre - post, pre, post),
+            _len => return Err(anyhow!("Prefix and suffix overlap, which is impossible")),
+        };
+
+        let mut span = Span::from(literal.span());
+
+        // check if it is a `///` comment, for which the literal
+        // span needs to be adjusted, since it would include the `///`
+        // @todo find a better way, potentially doing this when
+        // creating a `TrimmedLiteral` and storing this on construction
+        if pre == 1 && span.start.column == 0 {
+            span.start.column += 2;
+        }
+        span.start.column += pre;
+        span.end.column -= post;
+
+        Ok(Self {
+            len,
+            rendered,
+            span,
+            pre,
+            post,
+        })
+    }
+}
+
+impl TrimmedLiteral {
+    pub fn as_str(&self) -> &str {
+        &self.rendered.as_str()[self.pre..(self.pre + self.len)]
+    }
+    pub fn prefix(&self) -> &str {
+        &self.rendered.as_str()[..self.pre]
+    }
+    pub fn suffix(&self) -> &str {
+        &self.rendered.as_str()[(self.pre + self.len)..]
+    }
+
+    pub fn as_untrimmed_str(&self) -> &str {
+        &self.rendered.as_str()
+    }
+
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn pre(&self) -> usize {
+        self.pre
+    }
+
+    pub fn post(&self) -> usize {
+        self.post
+    }
+
+    pub fn span(&self) -> Span {
+        self.span.clone()
+    }
+
+    pub(crate) fn display(&self, highlight: Range) -> TrimmedLiteralDisplay {
+        TrimmedLiteralDisplay::from((self, highlight))
+    }
+}
+
+impl fmt::Debug for TrimmedLiteral {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use console::Style;
+
+        let pick = Style::new().on_black().underlined().dim().cyan();
+        let cutoff = Style::new().on_black().bold().dim().yellow();
+
+        write!(
+            formatter,
+            "{}{}{}",
+            cutoff.apply_to(&self.rendered.as_str()[0..self.pre]),
+            pick.apply_to(&self.rendered.as_str()[self.pre..(self.pre + self.len)]),
+            cutoff.apply_to(&self.rendered.as_str()[(self.pre + self.len)..]),
```

Comment:
use the corresponding functions instead of juggling with the offsets and lengths?

```suggestion
            cutoff.apply_to(&self.prefix()),
            pick.apply_to(&self.as_str()),
            cutoff.apply_to(&self.suffix()),
```